iothub_client/src/iothub_client_core.c

```diff
@@ -1691,13 +1699,21 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_SetOption(IOTHUB_CLIENT_CORE_HANDLE iotHub
         }
         else
         {
-            /*Codes_SRS_IOTHUBCLIENT_02_038: [If optionName doesn't match one of the options handled by this module then IoTHubClient_SetOption shall call IoTHubClientCore_LL_SetOption passing the same parameters and return what IoTHubClientCore_LL_SetOption returns.] */
-            result = IoTHubClientCore_LL_SetOption(iotHubClientInstance->IoTHubClientLLHandle, optionName, value);
-            if (result != IOTHUB_CLIENT_OK)
+            /* Codes_SRS_IOTHUBCLIENT_41_001 [ If parameter `optionName` is `OPTION_CONVENIENCE_LOOP_TIME` then `IoTHubClient_SetOption` shall set `loop_timeout` parameter of `IoTHubClientInstance` ]*/ 
+            if (strcmp(OPTION_CONVENIENCE_LOOP_TIME, optionName) == 0)
             {
-                LogError("IoTHubClientCore_LL_SetOption failed");
+                iotHubClientInstance->loop_timeout = *((uint16_t *)value);
```

Comment:
>iotHubClientInstance->loop_timeout = *((uint16_t *)value); [](start = 16, length = 58)

This set cannot happen here. It needs to occur within the LL DoWork itself because a race condition can occur in ScheduleWorkThread function around the read on low-end devices. There is no lock around the Sleep but there is one around the LL DoWork itself.

Reply:
This comment is not relevant in this scenario since we are threadlocked here, but makes sense for when I retrieve the value for ThreadAPI_Sleep. I've fixed that.